CMakeLists.txt

```diff
@@ -469,68 +469,81 @@ foreach(e ${LIBVNCCLIENT_EXAMPLES})
   target_link_libraries(client_examples_${e} vncclient ${CMAKE_THREAD_LIBS_INIT} ${SDL_LIBRARY} ${FFMPEG_LIBRARIES})
 endforeach(e ${LIBVNCCLIENT_EXAMPLES})
 
-
 #
 # them tests
 #
-
 if(UNIX)
     set(ADDITIONAL_TEST_LIBS m)
 endif(UNIX)
 
 set(SIMPLETESTS
    cargstest
    copyrecttest
+   wstest
 )
 
-if(CMAKE_USE_PTHREADS_INIT)
-  set(SIMPLETESTS
-      ${SIMPLETESTS}
-      encodingstest
+add_test(NAME cargs COMMAND test_cargstest)
+add_test(NAME websockets_decode COMMAND test_wstest)
+
+if(CMAKE_USE_PTHREADS_INI)
+  list(APPEND SIMPLETESTS encodingstest)
```

Comment:
this is admittedly nicer, but why did you touch that part that has nothing to do with websockets? ;-)